dali/kernels/signal/moving_mean_square.cc

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/signal/moving_mean_square.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+namespace {
+
+template<typename Result = float, typename T>
+float Square(const T &val) {
+  Result res = val;
+  return res * res;
+}
+
+
+template<typename T>
+float CalcSumSquared(span<const T> buffer, int start, int length) {
+  DALI_ENFORCE(buffer.size() >= length + start,
+               make_string_delim(" ", "Buffer overflow (size:", buffer.size(), "length:", length,
+                                 "start:", start));
+  float sumsq = 0;
+  for (int i = start; i < length + start; i++) {
```

Comment:
```suggestion
  for (int i = start; i < start + length; i++) {
```
would read better IMO